spec/features/visitors/email_confirmation_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+feature 'Email confirmation during sign up' do
+  scenario 'confirms valid email and sets valid password' do
+    email = 'test@example.com'
+    sign_up_with(email)
+    raw_confirmation_token, = Devise.token_generator.generate(User, :confirmation_token)
+    User.find_by(email: email).update(
+      confirmation_token: raw_confirmation_token, confirmation_sent_at: Time.current
+    )
+
+    visit "/users/confirmation?confirmation_token=#{raw_confirmation_token}"
+
```

Comment:
Since the confirmation is the core part of this spec rather than just setup, I thought it was nice to have in here explicitly rather than as part of a method that is in a different file and therefore not immediately visible. Does that make sense?